cpp/src/parquet/encryption/internal_file_decryptor.cc

```diff
@@ -64,17 +66,20 @@ InternalFileDecryptor::InternalFileDecryptor(FileDecryptionProperties* propertie
   properties_->set_utilized();
 }
 
+InternalFileDecryptor::~InternalFileDecryptor() { WipeOutDecryptionKeys(); }
+
 void InternalFileDecryptor::WipeOutDecryptionKeys() {
-  std::lock_guard<std::mutex> lock(mutex_);
+  std::unique_lock lock(mutex_);
   properties_->WipeOutDecryptionKeys();
```

Comment:
@pitrou Variable `properties` is shared across decryptor instances, wiping out keys mutates `properties` for other decryptors. Firstly, we revoke keys for other decryptors. Secondly, multiple decryptors might call `WipeOutDecryptionKeys` concurrently causing issues. A lock here does not help as individual decryptors have individual locks.

The keys should be wiped out in the destructor of `properties`, so they exist for the lifetime of `properties`.